frontend/controller/actions/group.js

```diff
@@ -262,15 +263,15 @@ export default (sbp('sbp/selectors/register', {
     // JOINING_GROUP, so that we process leave actions and don't interfere
     // with the leaving process (otherwise, the side-effects will prevent
     // us from fully leaving).
-    sbp('chelonia/contract/wait', [params.originatingContractID, params.contractID])
+    await sbp('chelonia/contract/wait', [params.originatingContractID, params.contractID])
     sbp('okTurtles.data/set', 'JOINING_GROUP-' + params.contractID, true)
     try {
       const { loggedIn } = sbp('state/vuex/state')
       if (!loggedIn) throw new Error('[gi.actions/group/join] Not logged in')
 
-      const { username, identityContractID: userID } = loggedIn
+      const { identityContractID: userID } = loggedIn
 
-      await sbp('chelonia/contract/sync', params.contractID)
+      await sbp('chelonia/contract/sync', params.contractID, { deferredRemove: true })
```

Comment:
This `deferredRemoved` has been added with no explanation. Please add a comment

Reply:
Done.